app/javascript/shared/components/ResizableTextArea.vue

```diff
@@ -34,18 +35,14 @@ export default {
   },
   watch: {
     value() {
-      this.resizeTextarea();
-    },
-  },
-  mounted() {
-    this.$nextTick(() => {
-      if (this.value) {
+      this.$nextTick(() => {
         this.resizeTextarea();
-      }
-    });
+      });
+    },
   },
   methods: {
     resizeTextarea() {
+      this.$el.style.height = 'inherit';
```

Comment:
Updating the DOM directly doesn't seem like a good practice. @scmmishra Thoughts?

Reply:
Let's use a computed property that returns the `height` and the style prop can be set directly

Reply:
I tried making this a computed property. But it does not get updated as we need. Any other ways you can think? Or we need to stick with this implementation 
<img width="454" alt="image" src="https://github.com/chatwoot/chatwoot/assets/1277421/fc4d938c-6b87-45bb-a456-b0a04f7da888">

Reply:
@pranavrajs @scmmishra

Reply:
Can we just return the value and see?

Reply:
Another thing is that the height: 'inherit' will always be overriden here

Reply:
@scmmishra Tried returning the value and keeping the inherit before it as the default. This isn't working either. 
<img width="470" alt="image" src="https://github.com/chatwoot/chatwoot/assets/1277421/1acd49d6-1885-4ab1-947d-1450b0c35b8d">

tried debounce function on the method also, but it isn't working as expected.

Reply:
Hey @nithindavid I tried rewriting the computed method as follows

```js
textAreaHeight() {
  if (!this.value) {
    return `${this.minHeight}rem`;
  }
  return this.$refs.textarea.scrollHeight + 'px';
}
```

and added the style prop as follows

```
:style="{
      height: textAreaHeight,
    }"
```

Seems to be working fine after that, can you give it a shot in case I missed something

<details>

<summary>
Full Code
</summary>

```vue
<template>
  <textarea
    ref="textarea"
    :placeholder="placeholder"
    :value="value"
    :style="{
      height: textAreaHeight,
    }"
    rows="1"
    @input="onInput"
    @focus="onFocus"
    @keyup="onKeyup"
    @blur="onBlur"
  />
</template>

<script>
const TYPING_INDICATOR_IDLE_TIME = 4000;
export default {
  props: {
    placeholder: {
      type: String,
      default: '',
    },
    value: {
      type: String,
      default: '',
    },
    minHeight: {
      type: Number,
      default: 2,
    },
  },
  data() {
    return {
      idleTimer: null,
    };
  },
  computed: {
    textAreaHeight() {
      if (!this.value) {
        return `${this.minHeight}rem`;
      }
      return this.$refs.textarea.scrollHeight + 'px';
    },
  },
  methods: {
    onInput(event) {
      this.$emit('input', event.target.value);
    },
    resetTyping() {
      this.$emit('typing-off');
      this.idleTimer = null;
    },
    turnOffIdleTimer() {
      if (this.idleTimer) {
        clearTimeout(this.idleTimer);
      }
    },
    onKeyup() {
      if (!this.idleTimer) {
        this.$emit('typing-on');
      }
      this.turnOffIdleTimer();
      this.idleTimer = setTimeout(
        () => this.resetTyping(),
        TYPING_INDICATOR_IDLE_TIME
      );
    },
    onBlur() {
      this.turnOffIdleTimer();
      this.resetTyping();
      this.$emit('blur');
    },
    onFocus() {
      this.$emit('focus');
    },
    focus() {
      this.$refs.textarea.focus();
    },
  },
};
</script>

```
</details>

Reply:
@scmmishra @pranavrajs Here's the case where the editor isn't set with the correct height. This is with the suggested solution. 

https://www.loom.com/share/83dd7a25a198464e83b2b611fd72923d

Reply:
@pranavrajs, with the current implementation the `resizeTextarea()` is called once for reloads and value changes. Also, not using tailwindcss as we have to use `h-[inherit]` which is getting overridden by lots of our existing styles.